base/statistics.jl

```diff
@@ -9,7 +9,7 @@ function mean(f::Callable, iterable)
     end
     count = 1
     value, state = next(iterable, state)
-    total = f(value)
```

Comment:
It almost certainly could but that's not the issue though. I thought this function applies to many other `f`s.